plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -175,6 +189,128 @@ fu_genesys_usbhub_device_read_flash(FuGenesysUsbhubDevice *self,
 	return TRUE;
 }
 
+static gboolean
+fu_genesys_usbhub_device_compare_flash_blank(FuGenesysUsbhubDevice *self,
+					     guint start_addr,
+					     guint code_size,
+					     FuProgress *progress,
+					     GError **error)
+{
+	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(self));
+	guint read_addr = start_addr + code_size - 0x400;
+	guint read_size = 0x400;
+	g_autoptr(GPtrArray) chunks = NULL;
+	g_autofree guint8 *blank_buf = NULL;
+	g_autofree guint8 *buf = NULL;
+
+	buf = g_malloc(self->flash_rw_size);
+	blank_buf = g_malloc(self->flash_rw_size);
+	memset(blank_buf, 0xff, self->flash_rw_size);
```

Comment:
could you use `g_autoptr(GByteArray) buf = g_byte_array_new()` and `fu_byte_array_set_size(buf, self->flash_rw_size, 0xFF` here instead? I think you can move both inside the loop too.

Reply:
I'm reusing the buffer to read. Maybe keep them outside of the loop is better.

Reply:
Sure, I'm fine with that -- but I think GByteArray is a bit more elegant.